### ISSUE TITLE
Allow app to turn off printing errors to logcat

### ISSUE THREAD

**Issue by noamtamim** [google/ExoPlayer#4665]
### Issue description
When there are playback errors, ExoPlayer sends them to logcat in addition to reporting via listeners:

```java
    // Snippet from ExoPlayerImplInternal
    } catch (ExoPlaybackException e) {
      Log.e(TAG, "Playback error.", e);
      stopInternal(/* reset= */ false, /* acknowledgeStop= */ false);
      eventHandler.obtainMessage(MSG_ERROR, e).sendToTarget();
      maybeNotifyPlaybackInfoChanged();
    } catch (IOException e) {
      Log.e(TAG, "Source error.", e);
      stopInternal(/* reset= */ false, /* acknowledgeStop= */ false);
      eventHandler.obtainMessage(MSG_ERROR, ExoPlaybackException.createForSource(e)).sendToTarget();
      maybeNotifyPlaybackInfoChanged();
    } catch (RuntimeException e) {
      Log.e(TAG, "Internal runtime error.", e);
      stopInternal(/* reset= */ false, /* acknowledgeStop= */ false);
      eventHandler.obtainMessage(MSG_ERROR, ExoPlaybackException.createForUnexpected(e))
          .sendToTarget();
      maybeNotifyPlaybackInfoChanged();
```

I believe the player should not print to logcat. The app listens to `onPlayerError()` and it can decide to log these errors according to its own logic.

### Version of ExoPlayer being used
2.8.3

**Comment by ojw28** [comment]
This is a valid point, but on the flip side it's helpful for us to have some logging enabled by default, so that when less advanced developers file "playback doesn't work" type issues, there's something in the logs to go on.

We could leave the logging on by default and add a way of disabling it. Is it really important or worth it though? Is the logging causing any non-trivial issues for you? Thanks!

**Comment by noamtamim** [comment]
Upon startup, we run a test (see #3835) that always causes ExoPlayer to fail (the failure error is what matters for the purpose of the test). We don't want this to be printed to the logcat because it looks like something bad has happened.
Further, it seems that some (all?) of the crash-reporting services also report logcat errors. This causes a lot of false positives in the reports.

So yes, an ability to turn it off will be great. And I agree that leaving it on by default is probably a good thing.

**Comment by noamtamim** [comment]
Hi @ojw28, appreciate your response here.

**Comment by ojw28** [comment]
Ack. We'll probably need to redirect all `Log.*` calls through our own class to do this in a sensible way.

**Comment by noamtamim** [comment]
@ojw28 I believe this can be closed with the release of 2.9.0, right?

**Comment by ojw28** [comment]
Believe so; thanks!
